### ISSUE TITLE
bug: inline comments are destroyed

### ISSUE THREAD

**Issue by wongjiahau** [andrewbaxter/genemichaels#89]
My code (trimmed):
```rs
    fn get_valid_selection(
        get_actual_edit_transaction: impl Fn(
            /* current */ &Selection,
            /* next */ &Selection,
        ) -> anyhow::Result<EditTransaction>,
    ) -> anyhow::Result<Either<Selection, EditTransaction>> {
        todo!()
    }
```

Error (trimmed):
```
 Error formatting file
    - file = xxx
    Caused by:
      Error doing formatting
        Caused by:
          Encountered a bug; some comments were lost during formatting
            - comments = [Whitespace { loc: LineColumn { line: 1530, column: 26 }, mode: Comment(Comment { mode: Normal, lines: "current" }) }, Whitespace { loc: LineColumn
                         { line: 1531, column: 23 }, mode: Comment(Comment { mode: Normal, lines: "next" }) }]
```

**Comment by wongjiahau** [comment]
By the way, I just want to express my appreciation of this project, because I'm sick of `cargo fmt` giving up on lines that are too long.

**Comment by andrewbaxter** [comment]
I think I've released this (0.5.7).  To summarize, I need to check+add comments before every symbol in the AST and missed it before `&` here.  There's a test now so at least hopefully there won't be regressions.
